### ISSUE TITLE
Misplaced gutter icons

### ISSUE THREAD

**Issue by TeeSeal** [facebook-atom/atom-ide-ui#108]
### Description
Gutter icons from the diagnostics module pop up to the left of line numbers. In fact, while no icons are visible it looks fine, as soon as a warning pops up, the whole line number row is moved to the right to make space for the icons. Not super important, but I find the way the linter package does it to be way more pleasant to the eye.

### Expected Behavior
<img width="130" alt="screen shot 2017-10-31 at 18 07 47" src="https://user-images.githubusercontent.com/14080148/32234822-70f957ee-be66-11e7-8a29-3f6020cbd739.png">

### Actual Behavior
<img width="144" alt="screen shot 2017-10-31 at 18 02 40" src="https://user-images.githubusercontent.com/14080148/32234750-49ffa170-be66-11e7-8761-afd2106c5681.png">

### Versions
* Atom: 1.21.1
* Client OS: macOS High Sierra 10.13
* atom-ide-ui: 0.5.3

**Comment by Arcanemagus** [comment]
Just to note, the side of the line numbers that the `linter-ui-default` shows the indicators on is configurable.

**Comment by hansonw** [comment]
Yup, seems like a great configuration option / PR candidate! We've made this the default to play nicely with Nuclide's debugger functionality, which also puts breakpoints on the left - but I definitely see how the right side could be better for you. The gutters are created on https://github.com/facebook-atom/atom-ide-ui/blob/master/modules/atom-ide-ui/pkg/atom-ide-diagnostics-ui/lib/gutter.js#L83.

**Comment by TeeSeal** [comment]
Well, yeah `linter-ui-default` can be configured to place the icons on the left, but space for those icons is reserved from the get go, so they just appear in the reserved space.

With diagnostics however there is no space reserved from the beginning, and when it is needed it just changes the UI to make spaces, which weirds me out.

Even if that is fixed, it would still be nice to have an option to put the icons on the right.

**Comment by Arcanemagus** [comment]
> With diagnostics however there is no space reserved from the beginning, and when it is needed it just changes the UI to make space, which weirds me out.

Ah, yea that's something that was fixed pretty early in `linter-ui-default`'s development. A _lot_ of people don't like their UI jumping around on them 😛.

**Comment by phaux** [comment]
I changed the default behavior using custom style:

```css
.gutter[gutter-name=diagnostics-gutter] {
  order: 1;
}
```
